content/docs/08.architecture.md

```diff
@@ -61,26 +61,24 @@ You can scale Executors as necessary, but as no heavy computations are done in t
 
 ### Worker
 
-The **Worker** is where all tasks are run. A Worker will receive a task from the Executor and will process it. As tasks can be nearly anything (heavy computations, simple API calls, etc.), the Worker is essentially a Thread Pool where you can configure the number of threads you need.
+The **Worker** is where all tasks and polling triggers are run. A Worker will receive a task from the Executor or a polling trigger from the Scheduler and will process it. As tasks and polling triggers can be nearly anything (heavy computations, simple API calls, etc.), the Worker is essentially a Thread Pool where you can configure the number of threads you need.
 
 You can scale Workers as necessary and have many instances on multiple servers, each with its own Thread Pool.
 
-As the worker will execute the tasks, it needs access to any external services used by your tasks (database, REST services, message broker, etc.).
+As the worker executes the tasks and the polling triggers, it needs access to any external services they use (database, REST services, message broker, etc.). Workers are the only server components that need access to external services.
 
 ### Scheduler
 
 The **Scheduler** will handle most of the [Triggers](./05.developer-guide/08.triggers/index.md) (except the [Flow Triggers handled by the Executor](./05.developer-guide/08.triggers/02.flow.md)). It will continuously watch all triggers and, if all the conditions are met, will trigger an execution (submit it to the Executor).
```

Comment:
```suggestion
The **Scheduler** will handle most of the [Triggers](./05.developer-guide/08.triggers/index.md) except the [Flow Triggers handled by the Executor](./05.developer-guide/08.triggers/02.flow.md). It will continuously watch all triggers and, if all conditions are met, will execute the flow (submit the flow to the Executor).
```

triggers vs. "to trigger" as synonym to "run/execute" can be confusing

Reply:
The Scheduler will not execute the flow but trigger an execution.
The Flow is executed by the executor and the worker.

Maybe we can replace by '..., will start an execution (submit ...)